src/python/pants/backend/python/tasks/python_task.py

```diff
@@ -187,3 +192,12 @@ def _chroot_path(self, interpreter, pex_info, targets, platforms, extra_requirem
 
     fingerprint = hash_utils.hash_all(fingerprint_components)
     return os.path.join(self.chroot_cache_dir, fingerprint)
+
+  def _configure_shebang(self, targets, path, interpreter, pex_info):
+    builder = PEXBuilder(path=path, interpreter=interpreter, pex_info=pex_info, copy=True)
+
+    binary_targets = [target for target in targets if isinstance(target, PythonBinary)]
+    if len(binary_targets) == 1 and binary_targets[0].shebang:
+      builder.set_shebang(binary_targets[0].shebang)
```

Comment:
should this `logger.warn` if `len(binary_targets) > 1` so users know the shebang won't be applied in that case?

Reply:
I think we're in agreement that these changes to the old non-default pipeline can be reverted. If not, I'd want to double-down on @kwlzn's suggestion and fail fast with a `TaskError` raise. In other words, `shebang` compatible with the v1 pipeline in limited circumstances (non-batch `binary` builds).